modules/user/user.repository.js

```diff
@@ -1,35 +1,84 @@
 const mongoose = require("mongoose");
 const Schema = mongoose.Schema;
 const { CustomError } = require("../../errorHandler");
-require("dotenv").config();
+const { encodeJWT } = require("./jwt.config");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
 
 const userScheme = new Schema({
-    id: Number,
-    name: String,
-    surname: String,
-    email: String,
-    phone: Number,
+    id: {
+        type: Number,
+        required: true,
+        unique: true,
+    },
+    name: {
+        type: String,
+        required: true,
+    },
+    surname: {
+        type: String,
+        required: true,
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    phone: {
+        type: Number,
+        required: true,
+        unique: true,
+    },
+    password: {
+        type: String,
+        required: true,
+    },
 });
 const UserModel = mongoose.model("User", userScheme);
 
 const userAggregateForm = {
-    _id: 0,
-    id: 1,
-    name: 1,
-    surname: 1,
-    email: 1,
-    phone: 1,
+    _id: false,
+    id: true,
+    name: true,
+    surname: true,
+    email: true,
+    phone: true,
+    password: true,
 };
 
 class UserRepository {
+    async userLogin(userCred) {
+        const userDB = await UserModel.findOne(
+            { email: userCred.email },
+            userAggregateForm
+        );
+
+        if (!userDB) {
+            throw new CustomError("EMAIL_IS_INCORRECT");
+        }
+
+        const matchLoginPass = await bcrypt.compare(
+            userCred.password,
+            userDB.password
+        );
+
+        if (matchLoginPass) {
```

Comment:
This If statement is going to be cleaner if we will change it to 
`if (!matchLoginPass) {
throw Err()
}`

- [ ] change if statement